mermithid/misc/FakeTritiumDataFunctions.py

```diff
@@ -0,0 +1,655 @@
+        np.save('scatter_spectra_files/scatter'+gas_type+"_"+str(i).zfill(2),current_scatter)
+    # plt.plot(x,current_scatter) # diagnostic
+    # plt.show() # diagnostic
+    elapsed = time.time() - t
+    logger.info('Files generated in '+str(elapsed)+'s')
+    return
+
+# Returns the name of the current path
+def get_current_path():
+    path = os.path.abspath(os.getcwd())
+    return path
+
+# Prints a list of the contents of a directory
+def list_files(path):
+    directory = os.popen("ls "+path).readlines()
+    strippeddirs = [s.strip('\n') for s in directory]
+    return strippeddirs
+
+# Returns the name of the current directory
+def get_current_dir():
+    current_path = os.popen("pwd").readlines()
+    stripped_path = [s.strip('\n') for s in current_path]
+    stripped_path = stripped_path[0].split('/')
+    current_dir = stripped_path[len(stripped_path)-1]
+    return current_dir
+
+# Checks for the existence of a directory called 'scatter_spectra_files'
+# and checks that this directory contains the scatter spectra files.
+# If not, this function calls generate_scatter_convolution_files.
+# This function also checks to make sure that the scatter files have the correct
+# number of points in the SELA, and if not, it generates fresh files
+def check_existence_of_scatter_files(gas_type):
+    current_path = get_current_path()
+    current_dir = get_current_dir()
+    stuff_in_dir = list_files(current_path)
+    if 'scatter_spectra_files' not in stuff_in_dir and current_dir != 'scatter_spectra_files':
+        logger.warning('Scatter files not found, generating')
+        os.popen("mkdir scatter_spectra_files")
+        time.sleep(2)
+        generate_scatter_convolution_files(gas_type)
+    else:
+        directory = os.popen("ls scatter_spectra_files").readlines()
+        strippeddirs = [s.strip('\n') for s in directory]
+        if len(directory) != len(gases) * max_scatters:
+            generate_scatter_convolution_files(gas_type)
+        test_file = 'scatter_spectra_files/scatter'+gas_type+'_01.npy'
+        test_arr = np.load(test_file)
+        if len(test_arr) != num_points_in_std_array:
+            logger.warning('Scatter files do not match standard array binning, generating fresh files')
+            generate_scatter_convolution_files(gas_type)
+    return
+
+# Given a function evaluated on the SELA, convolves it with a gaussian
+def convolve_gaussian(func_to_convolve,gauss_FWHM_eV):
+    sigma = gaussian_FWHM_to_sigma(gauss_FWHM_eV)
+    resolution_f = std_gaussian(sigma)
+    ans = convolve(resolution_f,func_to_convolve,mode='same')
+    ans_normed = normalize(ans)
+    return ans_normed
+
+# Produces a full spectral shape on the SELA, given a gaussian resolution
+# and a scatter probability
+def make_spectrum(gauss_FWHM_eV,scatter_prob,gas_type, emitted_peak='lorentzian'):
+    current_path = get_current_path()
+    # check_existence_of_scatter_files()
```

Comment:
This function seems useful in general: it should be used in the Configure of the processor and make sure the files are present before running the processor on the data...

Reply:
fixing this is too painful right now. the hope is that the detailed lineshape branch will make this and the other detailed lineshape functions redundant so they can be removed here. Until then, I added a check for the path existence in the Configure method of the processor.